.github/workflows/native-wsl.yml

```diff
@@ -0,0 +1,175 @@
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    defaults:
+        run:
+          shell: ${{ matrix.configuration == 'wsl' && 'wsl-bash {0}' || 'pwsh' }}
+    strategy:
+      matrix:
+        os: [windows-latest]
+        node-version: [16, 14, 12, 10]
+        configuration: [wsl, native]
+        architecture: [x86, x64]
+    steps:
+    - uses: actions/checkout@v4
+    - uses: Vampire/setup-wsl@v3
+      if: matrix.configuration == 'wsl'
+      with:
+        distribution: Ubuntu-22.04
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      if: matrix.configuration == 'native'
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Do everything here because npm is not found when running in separate steps
+      if: matrix.configuration == 'wsl'
+      run: |
+        export ESLINT_VERSION=7
+        sudo apt update -y
+        sudo apt-get install curl -y
+        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+        export NVM_DIR="$HOME/.nvm"
+        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+        [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+        nvm install ${{ matrix.node-version }}
+        nvm use ${{ matrix.node-version }}
+        node --version
+        if [ ${{ matrix.node-version }} -lt 11 ]; then
+          npm install --legacy-peer-deps
+        fi
+        if [ ${{ matrix.node-version }} -gt 10 ]; then
+          npm install
+        fi
+        npm install nyc@latest --no-save
+        set TRAVIS_NODE_VERSION=${{ matrix.node-version }}
+        if [ ${{ matrix.node-version }} -le 8 ]; then
+          set ESLINT_VERSION=6
+        fi
+        if [ ${{ matrix.node-version }} -gt 8 ]; then
+          set ESLINT_VERSION=7
+        fi
+        if [ ${{ matrix.node-version }} -ge 11 ]; then
+          set ESLINT_VERSION=8
+        fi
+        npm run copy-metafiles
+        # bash ./tests/dep-time-travel.sh 2>&1
+        npm install eslint@${ESLINT_VERSION} --no-save --ignore-scripts
+        if [ ${{ matrix.node-version }} -lt 12 ]; then
+          npm install @typescript-eslint/parser@4
+        fi
+        if [ ESLINT_VERSION -eq 8 ]; then
+          npm i --no-save eslint-plugin-import@'.' -f
+          npm run build
+        fi
+
+        git config core.symlinks true
+        git reset --hard
+        npm config set package-lock false
+        resolverDir="./resolvers"
+        for resolver in $resolverDir/*; do
+          if [ -d "$resolver" ]; then  # Add this line to check if $resolver is a directory
+            pushd $resolver
+            npm install
+            npm ls nyc > /dev/null
+            if [ $? -eq 0 ]; then
+              npm install nyc@latest --no-save
+            fi
+            popd
+          fi  # Close the if statement here
+        done
+        npm config set package-lock true
```

Comment:
meaning that this isn't needed either - i don't use lockfiles in packages

Reply:
I believe what's happening here is that there's mixed/incorrect versions. Here's the error happening in `posttest`

```sh
Run npm run posttest
  npm run posttest
  shell: /usr/bin/bash -e {0}

> eslint-plugin-import@2.29.1 posttest
> eslint . && npm run update:eslint-docs -- --check && markdownlint "**/*.md"


/home/runner/work/eslint-plugin-import/eslint-plugin-import/tests/src/core/resolve.js
  116:95  error  A space is required before '}'  object-curly-spacing

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

Error: Process completed with exit code 1.
```

Setting the `package-lock` to false prior to installing in this run fixes that error. I set it back to true just because I want to "reset" what is down at the beginning.

Reply:
that's very confusing since it should be false for the entire repo, at all times :-/